components/base/kinematicbase/fake_kinematics.go

```diff
@@ -187,25 +192,36 @@ func WrapWithFakePTGKinematics(
 		return nil, err
 	}
 
+	// construct execution frame
+	executionFrame, err := referenceframe.New2DMobileModelFrame(
+		b.Name().ShortName()+"ExecutionFrame",
+		planningFrame.DoF(),
+		geometries[0],
```

Comment:
[q]: for PTG frames we say that they may be defined as a slice of geometries but for the 2D mobileFrame we only accept a singular geometry... This should probably be updated such that there is uniformity.